src/Metadata/Property/Factory/AnnotationSubresourceMetadataFactory.php

```diff
@@ -70,16 +71,19 @@ public function create(string $resourceClass, string $property, array $options =
             $annotation = $this->reader->getMethodAnnotation($reflectionMethod, ApiSubresource::class);
 
             if (null !== $annotation) {
-                return $this->updateMetadata($annotation, $propertyMetadata, $resourceClass);
+                return $this->updateMetadata($annotation, $propertyMetadata, $resourceClass, $property);
             }
         }
 
         return $propertyMetadata;
     }
 
-    private function updateMetadata(ApiSubresource $annotation, PropertyMetadata $propertyMetadata, string $originResourceClass): PropertyMetadata
+    private function updateMetadata(ApiSubresource $annotation, PropertyMetadata $propertyMetadata, string $originResourceClass, string $propertyName): PropertyMetadata
     {
         $type = $propertyMetadata->getType();
+        if (null === $type) {
+            throw new InvalidResourceException(sprintf('Property %s on resource %s has no type defined, but is declared as a subresource.', $propertyName, $originResourceClass));
```

Comment:
I think saying that it "has no type defined" is confusing... Perhaps we could say:

Property %s on resource %s is declared as a subresource, but its type could not be determined.

Reply:
I love your suggestion indeed, your english is too good haha.

Reply:
Changed.